src/phase0/state_transition/epoch_processing.rs

```diff
@@ -315,8 +315,18 @@ pub fn process_historical_roots_update<
         MAX_VALIDATORS_PER_COMMITTEE,
         PENDING_ATTESTATIONS_BOUND,
     >,
-    _context: &Context,
+    context: &Context,
 ) -> Result<(), Error> {
+    let next_epoch = get_current_epoch(state, context) + 1;
+    if (next_epoch % (SLOTS_PER_HISTORICAL_ROOT as u64 / context.slots_per_epoch)) == 0 {
+        let mut historical_batch = HistoricalBatch {
+            block_roots: state.block_roots.clone(),
+            state_roots: state.state_roots.clone(),
```

Comment:
similarly to this: https://github.com/ralexstokes/ethereum_consensus/pull/36#discussion_r775979388

i'm going to want to avoid these `clone`s :)

if we do the most straightforward thing to remove the calls to `clone`, these fields then become references:

```rust
#[derive(Default, Debug, SimpleSerialize)]
pub struct HistoricalBatch<const SLOTS_PER_HISTORICAL_ROOT: usize> {
    pub block_roots: &Vector<Root, SLOTS_PER_HISTORICAL_ROOT>,
    pub state_roots: &Vector<Root, SLOTS_PER_HISTORICAL_ROOT>,
}
```

and we hit a problem as we can no longer derive `Default` or `SimpleSerialize`: there is no `Default` for `&T` as that would likely just point to "unsafe" uninitialized memory, and `SimpleSerialize` needs `Default` right now for the {de,}serialization routines.

so rather than make broad changes to `ssz_rs` at the moment (e.g. we could try to relax the `Default` constraint, which will be tricky, or we could pull out the merkleization functionality into a separate trait (but then need to implement `derive` functionality, etc.)), we can just leverage what we know about SSZ to work around this problem at the cost of a small deviation from the spec.

the `HistoricalBatch` container is essentially just a wrapper around the `block_roots` and `state_roots` data to get a nice "summary" merkle root over the entire set of block and state roots. and given that we just need this one root, we can basically perform part of the Merkleization manually and then use `ssz_rs` to do the rest. note: if we needed to serialize the `HistoricalBatch` data we could not take advantage of this but luckily we just need the root!

to finish the thought, we make some changes like this:

```rust
// 1. change definition
// and perhaps leave a helpful comment (or at least link this github comment!) to document why we deviate from the spec here
#[derive(Default, Debug, SimpleSerialize)]
pub struct HistoricalBatchAccumulator {
    pub block_roots_root: Root,
    pub state_roots_root: Root,
}

// 2. and then in this function:
let mut historical_batch = HistoricalBatchAccumulator {
    block_roots_root: state.block_roots.hash_tree_root()?,
    state_roots_root: state.state_roots.hash_tree_root()?,
};
```

Reply:
Got it -- makes sense. I've made the necessary changes and want to call out one minor casting needed in `Context` after the `slots_per_historical_root` type change...which I think is the desired approach but double checking: https://github.com/buchhlz2/ethereum_consensus/commit/6d0b56b1807ce82606dd1dbd9de2840a86112f99#r62609271